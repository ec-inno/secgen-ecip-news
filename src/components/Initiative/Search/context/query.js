import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
