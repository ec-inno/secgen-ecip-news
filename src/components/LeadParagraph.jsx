import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

const LeadParagraph = ({ intro, linkText, linkPath }) => (
  <div className="ecl-row ecl-u-mt-l">
    <div className="ecl-col-sm-12 ecl-col-md-12 ecl-u-mv-m">
      <p className="ecl-u-type-paragraph-lead">{intro}</p>
      <Link
        className="ecl-u-type-xl ecl-link ecl-link--standalone"
        to={linkPath}
      >
        {linkText}
      </Link>
    </div>
  </div>
);

LeadParagraph.propTypes = {
  intro: PropTypes.string,
  linkText: PropTypes.string,
  linkPath: PropTypes.string,
};

LeadParagraph.defaultProps = {
  intro: '',
  linkText: '',
  linkPath: '',
};

export default LeadParagraph;
