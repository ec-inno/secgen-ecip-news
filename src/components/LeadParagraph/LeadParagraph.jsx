import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

import Counter from '../Counter';
import Video from '../Video';

const LeadParagraph = ({ text, label, href }) => (
  <div className="ecl-row ecl-u-mt-l">
    <div className="ecl-col-sm-12 ecl-col-md-8 ecl-u-mv-m">
      <p className="ecl-u-type-paragraph-lead ecl-u-mt-none">{text}</p>
      <Link className="ecl-u-type-xl ecl-link ecl-link--standalone" to={href}>
        {label}
      </Link>
      <Counter />
    </div>
    <div className="ecl-col-sm-12 ecl-col-md-4 ecl-u-mv-m">
      <Video />
    </div>
  </div>
);

LeadParagraph.propTypes = {
  text: PropTypes.string,
  label: PropTypes.string,
  href: PropTypes.string,
};

LeadParagraph.defaultProps = {
  text: '',
  label: '',
  href: '',
};

export default LeadParagraph;
