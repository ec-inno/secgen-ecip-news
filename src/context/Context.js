import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
