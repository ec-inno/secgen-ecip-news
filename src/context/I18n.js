import React from 'react';

const I18n = React.createContext();

export default I18n;
