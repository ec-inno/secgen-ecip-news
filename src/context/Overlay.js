import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
