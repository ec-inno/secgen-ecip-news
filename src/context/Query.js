import React, { useContext } from 'react';

import { defaultLangKey } from '../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'unsetFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

const useQueryContext = () => useContext(Context);

export { queryInit, queryReducer, Context as default, useQueryContext };
