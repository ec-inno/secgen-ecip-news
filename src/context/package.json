{
  "name": "@eci/context",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@ecl/eu-preset-website": "2.8.0",
    "react": "16.9.0"
  }
}
