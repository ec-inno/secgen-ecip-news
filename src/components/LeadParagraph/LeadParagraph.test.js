import React from 'react';
import renderer from 'react-test-renderer';

import * as I18nContext from '@eci/context/I18n';
import LeadParagraph from './LeadParagraph';

describe('LeadParagraph', () => {
  it('renders correctly', () => {
    // `Counter` is part of the lead paragraph and needs this context to get `locale`
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: { pathname: '/en' },
    }));

    const props = { text: 'intro', label: 'Call to action', href: '#' };
    const tree = renderer.create(<LeadParagraph {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
