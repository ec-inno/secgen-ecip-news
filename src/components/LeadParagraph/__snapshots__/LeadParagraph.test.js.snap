// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadParagraph renders correctly 1`] = `
<div
  className="ecl-row ecl-u-mt-l"
>
  <div
    className="ecl-col-sm-12 ecl-col-md-8 ecl-u-mv-m"
  >
    <p
      className="ecl-u-type-paragraph-lead ecl-u-mt-none"
    >
      intro
    </p>
    <a
      className="ecl-u-type-xl ecl-link ecl-link--standalone"
      href="#"
    >
      Call to action
    </a>
    
  </div>
  <div
    className="ecl-col-sm-12 ecl-col-md-4 ecl-u-mv-m"
  >
    <figure
      className="ecl-media-container"
    >
      <img
        alt="The European Citizens' Initiative in video."
        className="ecl-media-container__media"
        src="file"
      />
      <figcaption
        className="ecl-media-container__caption"
      >
        The European Citizens' Initiative in video.
      </figcaption>
    </figure>
  </div>
</div>
`;
